### PR TITLE
Don't set billboard.image values if they are the same.

### PR DESCRIPTION
**Pull request by mramato**
Every time we assign a canvas to billboard.image it creates a new texture, even if it's the same canvas instance. This is by design because canvases can be drawn to multiple times. (like people trying to do billboard animation). Since using a canvas causes this to happen every frame in the BillboardVisualizer, we are constantly creating textures and adding them to the atlas.

To fix this in the short term, I simply treat canvas elements in the Entity API as if it's always the same image. This prevents the problem and the only "feature" we lose is to keep drawing to a canvas and have it automatically update in the billboard (which is probably the wrong way to do billboard animation anyway since it's incredibly inneficient).

When I have some free time, I'll look into adding a proper billboard animation demo to Sandcastle using the Entity API. We may even be able to handle animated GIFs this way.

This also allowed me to remove some code from GeoJSONDataSource.

### CODE REVIEW

File: Source/DataSources/BillboardVisualizer.js [hunk 1]
```diff
@@ -50,6 +50,7 @@ define([
     var EntityData = function(entity) {
         this.entity = entity;
         this.billboard = undefined;
+        this.textureValue = undefined;
```
Comment (pjcozzi) [inline Source/DataSources/BillboardVisualizer.js]: It's a shame we have to keep a reference to this.  This is something we avoid like the plague in the lower-level graphics.

Comment (mramato) [inline Source/DataSources/BillboardVisualizer.js]: Most of the time this is string and `BillboardCollection` actually keeps a reference to it already.  The only thing BillboardCollection doesn't keep a reference to is the canvas objects, which we will now do here for the sake of comparing it against the original value.  However, in order for the object to be here, it has to exist somewhere else as well, so we are not really adding any extra memory into the system at all.

Comment (pjcozzi) [inline Source/DataSources/BillboardVisualizer.js]: I guess it's not important here but we have a common use case in 3D models and imagery were we create an image/canvas, upload it to the GPU, and then drop the reference to it.  This is why the texture atlas resize is done by basically copying the previous texture (via rendering it to a framebuffer) - so the texture atlas does not need to keep a reference to all the images in system memory.

Comment (mramato) [inline Source/DataSources/BillboardVisualizer.js]: I totally agree that this isn't ideal, but I think in the short term it's the best we can do.  Most of the time it's just going to be a string.  We can revisit this after KML.

**Pull request merged**